cmd/entrypoint/main.go

```diff
@@ -78,7 +80,16 @@ func main() {
 		log.Printf("non-fatal error copying credentials: %q", err)
 	}
 
-	if err := e.Go(); err != nil {
+	// Add timeout to context if a non-zero timeout is specified for a step
+	ctx := context.Background()
+	timeoutContext, _ := time.ParseDuration(*timeout)
```

Comment:
is the other return value here an error? we probably want to fail if the specified timeout is invalid

(maybe all of this could also happen in a function which we can unit test?)

Reply:
Indeed the other return value is an error.

However, the validity of a specified timeout is already checked in [task_validation.go](https://github.com/tektoncd/pipeline/blob/1cfb4d554b94e525f5788ac597a629d0e1a080c2/pkg/apis/pipeline/v1beta1/task_validation.go#L193) which causes the webhook to deny the creation of a taskrun in case an invalid timeout is specified which is why it's not checked here again.

Would you still recommend I check it here again and/or possibly add a unit test?